credentialdigger/models/snippet_model/snippet_model.py

```diff
@@ -76,17 +77,24 @@ def analyze(self, discovery):
             # since either the snippet is empty or there is just one word
             return True
 
-        if len(data) == 2:
-            # No need to pre-process data since we have only two words
-            # Predict if the string `word1 + ' ' + word2` is a false positive
-            label = self.model.predict(
-                data[0] + ' ' + data[1])[0]  # 0=label, 1=probability
+        # Extract the committed secret
+        index_of_value = self._label_preprocess(data)
+        
+        # Classify as a 'Leak' if this is a private key.
+        if self._check_private_key(data):
+            return False
+
+        # We ignore snippets that look like regular phrases with
+        # no assignment
+        if not any(not c.isalnum() and c not in ' _.,?!/' for c in raw_data):
+            return True
```

Comment:
I don't understand this line. What are you checking? That there is at least 1 symbol, and that this symbol is not in `' _.,?!/'` ?

Reply:
I am checking if the snippet at hand is a normal English expression and cannot (in almost all cases) be containing a leak.
Example:
```python
# put your password here
pwd=""
```
Using that check, we will classify the comment as FP without feeding it to the classifier.
NB: I know that comments may contain leaks, but I am not excluding comments, am excluding sentences that do not usually fall under code snippets that may contain leaks.

A real example: 
```python
raw_data = 'This is a code snippet, it contains no password. words like api_token, pwd, pass will make it detectable via the regex scanner, but this check tell us not to bother with. We know for sure that it is an FP! will this make things faster when scanning? Yes! . __copyrights__!.\
    Source file is /src/fake.txt'
if not any(not c.isalnum() and c not in ' _.,?!/' for c in raw_data):
    print(True)
else:
    print(False)
```
This code will print(True) => FP without us feeding this to the classifier. (
I have used all the mentioned symbols within the `raw_data` string.